zerver/models/users.py

```diff
@@ -917,9 +917,8 @@ def remote_user_to_email(remote_user: str) -> str:
 post_save.connect(flush_user_profile, sender=UserProfile)
 
 
-@cache_with_key(user_profile_by_id_cache_key, timeout=3600 * 24 * 7)
-def get_user_profile_by_id(user_profile_id: int) -> UserProfile:
-    return UserProfile.objects.select_related(
+def get_user_base_query_set() -> list[str]:
```

Comment:
QuerySet is a different type; `_base_query_fields` would be appropriate as a name for this function. 

But probably even better would be this  function name (or more naturally, `_base_queryset`), but have return `the actual `.select_related()` object with type `QuerySet[UserProfile]`, since that's avoid unnecessary repeated code.